Src/Support/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -87,6 +87,28 @@ public static Task<GoogleCredential> GetApplicationDefaultAsync()
         }
 
         /// <summary>
+        /// Returns the Application Default Credentials (ambient credentials) for the application, synchronously.
+        /// </summary>
+        /// <remarks>
+        /// See <see cref="GetApplicationDefaultAsync"/> for more details on the Application Default Credentials. If you
+        /// are writing an application using asynchronous operations, it is encouraged that you use that method. However,
+        /// if you are writing an application using synchronous operations this method is safe to call, even from a thread
+        /// associated with a single-thread synchronization context such as a Windows Forms UI. The call will block until
+        /// the credentials are available (or an exception is thrown) but will not attempt re-entrancy within the
+        /// synchronization context. So for example, if you call this from a Windows Forms UI thread, the UI will freeze
+        /// until the method completes, but it will not deadlock.
+        /// </remarks>
+        /// <returns>The Application Default Credentials (ambient credentials) for the application.</returns>
+        public static GoogleCredential GetApplicationDefault()
+        {
+            // Everywhere we await a task internally, we use ConfigureAwait(false). Therefore even if this is called on
```

Comment:
Forgot to reply to the "what if we happen to omit ConfigureAwait?" part earlier - that would indeed be a bug. *If* we use this PR in some form, I should add a test (pre-merge) that tests that we don't rely on re-entrancy. (I'm not sure how easy that will be to test, but it shouldn't be *too* hard...) I'll only do that if we go down the route of adding the method though.